codes/c/chapter_backtracking/permutations_i.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * File: permutations_i.c
+ * Created Time: 2023-06-04
+ * Author: Gonglja (glj0@outlook.com)
+ */
+
+#include "../utils/common.h"
+
+/* 定义向量类型 */
+typedef struct Vector {
```

Comment:
Could we extract the vector to the public module? Now there are many duplicated code in different files.